.github/workflows/config.yml

```diff
@@ -0,0 +1,735 @@
+    - name: Conditionally checkout monorepo
+      if: steps.celo-monorepo-cache.outputs.cache-hit != 'true'
+      uses: actions/checkout@v3
+      with:
+        repository: celo-org/celo-monorepo
+        path: celo-monorepo
+        ref: ${{ env.MONOREPO_COMMIT }}
+    - name: Conditionally setup Node.js
+      if: steps.celo-monorepo-cache.outputs.cache-hit != 'true'
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ env.NODE_VERSION }}
+        cache: yarn
+        cache-dependency-path: celo-monorepo/yarn.lock
+    # Python is required for node-gyp rebuild
+    - name: Conditionally setup python
+      if: steps.celo-monorepo-cache.outputs.cache-hit != 'true'
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ env.PYTHON_VERSION }}
+    - name: Conditionally build monorepo
+      if: steps.celo-monorepo-cache.outputs.cache-hit != 'true'
+      run: |
+        cd celo-monorepo
+        yarn
+        yarn build
+        cd packages/protocol
+        yarn run build:sol
+    - name: Upload built system contracts
+      if: always()
+      uses: actions/upload-artifact@v3
+      with:
+        name: system-contracts
+        path: 'celo-monorepo/${{ env.CONTRACTS_BUILD_PATH }}'
+
+  go-modules:
+    name: Setup go
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+    - name: Setup go
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ env.GO_VERSION }}
+        check-latest: false
+        cache: true
+
+  check-imports:
+    name: Check imports
+    needs: go-modules
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+    - name: Setup golang
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ env.GO_VERSION }}
+        check-latest: false
+        cache: true
+    - name: Check imports to ensure we are using github.com/celo-org/celo-blockchain
+      run: ./scripts/check_imports.sh
+
+  lint:
+    name: Lint code
+    runs-on: ubuntu-latest
+    needs: go-modules
+
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+    - name: Setup golang
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ env.GO_VERSION }}
+        check-latest: false
+        cache: true
+    - name: Run linter
+      run: |
+        # so that the tee call at the end doesn't hide the real exit code
+        set -o pipefail
+        mkdir -p /tmp/test-results
+        go run build/ci.go ensure-linter
+        ./build/cache/golangci-lint-1.42.0-linux-amd64/golangci-lint run --config .golangci.yml --out-format junit-xml ./... | tee /tmp/test-results/go-lint-report.xml
+    - name: Upload linter report
+      uses: actions/upload-artifact@v3
+      with:
+        name: linter-report
+        path: /tmp/test-results
+
+  unit-tests:
+    name: Unit tests
+    runs-on: ubuntu-latest
```

Comment:
Defining permissions for the default github token will allow the job to write to the PR, (for codecoverage step).  This can be defined under unit-tests in line with name and runs-on

  permissions: 
      pull-requests: write

Reply:
nevermind it's commenting already, closing this.